src/com/vinnik/kate/MultithreadingSorter.java

```diff
@@ -0,0 +1,56 @@
+package com.vinnik.kate;
+
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+/** Class, which creates given number of threads, and measures time they need to sort random array. */
+public class MultithreadingSorter {
+
+    /** Creates given number of threads and executes them. */
+    public static void measureTimeOfSorting(int amountOfThreads) {
+        for (int i = 0; i < amountOfThreads; i++) {
+            ThreadSort thread = new ThreadSort(i);
+            thread.start();
+        }
+    }
+
+
+    private static class ThreadSort extends Thread {
+        private int[] numbersToSort;
+        private int key;
+
+        protected ThreadSort(int number) {
```

Comment:
In this scope, it's not really clear what does "number" mean. I'd do the following changes in the code:
1. rename the class variable `key` -> `identifier`
2. Change the constructor's argument name like this: `protected ThreadSort(final int identifier) { this. identifier = identifier; }`